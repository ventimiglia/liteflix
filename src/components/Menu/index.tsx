"use client";

import React, { ReactNode, useState } from "react";
import Image from "next/image";
import Modal from "../Modal";
import { motion, AnimatePresence } from "framer-motion";

const Menu = ({ children }: { children: ReactNode }) => {
  const [showMenu, setShowMenu] = useState(false);

  return (
    <>
      <button
        className="flex items-center gap-3 transition-transform hover:scale-125 focus:scale-125"
        onClick={() => setShowMenu(true)}
      >
        {children}
      </button>
      <AnimatePresence>
        {showMenu && (
          <motion.div
            initial={{ opacity: 0 }}
            animate={{ opacity: 1 }}
            exit={{ opacity: 0 }}
            className="h-full overflow-y-hidden overflow-x-hidden lg:overflow-y-auto fixed inset-0 z-50 w-full outline-none bg-gray-900/70 focus:outline-none"
          >
            <motion.section
              key="menu"
              initial={{ x: "100%" }}
              animate={{ x: "0%" }}
              exit={{ x: "100%" }}
              className="lg:absolute lg:right-0 lg:w-[40%] h-full p-6 lg:py-6 lg:px-20 bg-secondary"
            >
              <header className="flex justify-between w-full gap-10">
                <Menu>
                  <Image
                    src="/plus.svg"
                    alt="Menu"
                    width={26}
                    height={26}
                    className="rotate-45"
                    onClick={() => setShowMenu(false)}
                  />
                </Menu>
                <div className="flex flex-grow justify-center lg:justify-end">
                  <Image
                    src="/notification.svg"
                    alt="Notifications"
                    width={26}
                    height={26}
                    className="hidden lg:flex self-center w-auto h-auto"
                  />
                  <Image
                    src="/logo.svg"
                    alt="Liteflix"
                    width={98}
                    height={28}
                    className="w-auto h-auto self-center lg:hidden"
                  />
                </div>
                <button>
                  <Image
                    src="/avatar.svg"
                    alt="Profile"
                    width={36}
                    height={36}
                  />
                </button>
              </header>
              <nav className="py-16">
                <ul className="flex flex-col gap-10 text-left">
                  <li>
                    <a href="#">Inicio</a>
                  </li>
                  <li>
                    <a href="#">Series</a>
                  </li>
                  <li>
                    <a href="#">Películas</a>
                  </li>
                  <li>
                    <a href="#">Aregadas recientemente</a>
                  </li>
                  <li>
                    <a href="#">Populares</a>
                  </li>
                  <li>
                    <a href="#">Mis películas</a>
                  </li>
                  <li>
                    <a href="#">Mi lista</a>
                  </li>
                  <li className="my-6">
                    <Modal>
                      <Image
                        src="/plus.svg"
                        alt="Agregar película"
                        width={14}
                        height={14}
                        className="relative font-bold text-lg"
                      />
                      <b>AGREGAR PELÍCULA</b>
                    </Modal>
                  </li>
                  <li>
                    <a href="#">Cerrar sesión</a>
                  </li>
                </ul>
              </nav>
            </motion.section>
          </motion.div>
        )}
      </AnimatePresence>
    </>
  );
};

export default Menu;
