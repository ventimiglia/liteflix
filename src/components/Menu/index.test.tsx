import React from "react";
import { render, fireEvent, waitFor } from "@testing-library/react";
import Menu from ".";

jest.mock("next/navigation", () => ({
  useRouter() {
    return {
      pathname: "/",
    };
  },
}));

describe("Menu", () => {
  it("opens the menu when the button is clicked", () => {
    const children = "Open Menu"
    const { container, getByText, getByRole } = render(
      <Menu>{children}</Menu>
    );
    const menuButton = getByText(children);

    fireEvent.click(menuButton);

    const menu = getByRole("menu");
    expect(menu).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it("closes the menu when the close button is clicked", async () => {
    const { getByText, queryByRole, getByAltText } = render(<Menu>Open Menu</Menu>);
    const menuButton = getByText("Open Menu");

    fireEvent.click(menuButton);

    const closeButton = getByAltText("close menu");
    fireEvent.click(closeButton);

    await waitFor(() => {
      const menu = queryByRole("menu");
      expect(menu).not.toBeInTheDocument();
    });
  });
});