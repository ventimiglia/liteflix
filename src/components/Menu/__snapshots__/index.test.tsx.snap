// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu opens the menu when the button is clicked 1`] = `
<div>
  <button
    class="flex items-center gap-3 transition-transform hover:scale-125 focus:scale-125"
  >
    Open Menu
  </button>
  <div
    class="h-full overflow-y-hidden overflow-x-hidden lg:overflow-y-auto fixed inset-0 z-50 w-full outline-none bg-gray-900/70 focus:outline-none"
    role="menu"
    style="opacity: 0;"
  >
    <section
      class="lg:absolute lg:right-0 lg:w-[40%] h-full p-6 lg:py-6 lg:px-20 bg-secondary"
      style="transform: translateX(100%) translateZ(0);"
    >
      <header
        class="flex justify-between w-full gap-10"
      >
        <img
          alt="close menu"
          class="rotate-45 cursor-pointer transition-transform hover:scale-125"
          data-nimg="1"
          decoding="async"
          height="26"
          loading="lazy"
          src="/plus.svg"
          style="color: transparent;"
          width="26"
        />
        <div
          class="flex flex-grow justify-center lg:justify-end"
        >
          <img
            alt="Notifications"
            class="hidden lg:flex self-center w-auto h-auto cursor-pointer transition-transform hover:scale-125"
            data-nimg="1"
            decoding="async"
            height="26"
            loading="lazy"
            src="/notification.svg"
            style="color: transparent;"
            width="26"
          />
          <img
            alt="Liteflix"
            class="w-auto h-auto self-center lg:hidden"
            data-nimg="1"
            decoding="async"
            height="28"
            loading="lazy"
            src="/logo.svg"
            style="color: transparent;"
            width="98"
          />
        </div>
        <button>
          <img
            alt="Profile"
            class="cursor-pointer transition-transform hover:scale-125"
            data-nimg="1"
            decoding="async"
            height="36"
            loading="lazy"
            src="/avatar.svg"
            style="color: transparent;"
            width="36"
          />
        </button>
      </header>
      <nav
        class="py-16"
      >
        <ul
          class="flex flex-col gap-10 text-left"
        >
          <li>
            <a
              href="#"
            >
              Inicio
            </a>
          </li>
          <li>
            <a
              href="#"
            >
              Series
            </a>
          </li>
          <li>
            <a
              href="#"
            >
              Películas
            </a>
          </li>
          <li>
            <a
              href="#"
            >
              Aregadas recientemente
            </a>
          </li>
          <li>
            <a
              href="#"
            >
              Populares
            </a>
          </li>
          <li>
            <a
              href="#"
            >
              Mis películas
            </a>
          </li>
          <li>
            <a
              href="#"
            >
              Mi lista
            </a>
          </li>
          <li
            class="my-6"
          >
            <button
              class="flex items-center gap-3 transition-transform hover:scale-105 focus:scale-105"
            >
              <img
                alt="Agregar película"
                class="relative font-bold text-lg"
                data-nimg="1"
                decoding="async"
                height="14"
                loading="lazy"
                src="/plus.svg"
                style="color: transparent;"
                width="14"
              />
              <b>
                AGREGAR PELÍCULA
              </b>
            </button>
          </li>
          <li>
            <a
              href="#"
            >
              Cerrar sesión
            </a>
          </li>
        </ul>
      </nav>
    </section>
  </div>
</div>
`;
